src/SubsCache.js

```diff
@@ -0,0 +1,223 @@
+import { EJSON } from 'meteor/ejson'
+
+function hasCallbacks(args){
+  // this logic is copied from Meteor.subscribe found in
+  // https://github.com/meteor/meteor/blob/master/packages/ddp/livedata_connection.js
+  if (args.length) {
+    let lastArg = args[args.length-1];
+    return _.isFunction(lastArg) || (lastArg && _.any([lastArg.onReady, lastArg.onError, lastArg.onStop], _.isFunction));
+  }
+};
+
+function withoutCallbacks(args){
+  if (hasCallbacks(args)) {
+    return args.slice(0,args.length-1);
+  } else {
+    return args.slice();
+  }
+};
+
+function callbacksFromArgs(args){
+  if (hasCallbacks(args)) {
+    if (_.isFunction(args[args.length-1])) {
+      return {onReady: args[args.length-1]};
+    } else {
+      return args[args.length-1];
+    }
+  } else {
+    return {};
+  }
+};
+
+SubsCache = function(expireAfter, cacheLimit) {
+  var self = this;
+
+  this.expireAfter = expireAfter || 5;
+  this.cacheLimit = cacheLimit || 10;
+  this.cache = {};
+  this.allReady = new ReactiveVar(true);
+
+  SubsCache.caches.push(this);
+
+  this.ready = function() {
+    return this.allReady.get();
+  }
+
+  this.onReady = function(callback) {
+    Tracker.autorun(function(c) {
+      if (self.allReady.get()) {
+        c.stop();
+        return callback();
+      }
+    });
+  }
+
+  this.clear = function() {
+    return _.values(this.cache).map(function(sub) { sub.stopNow() });
+  }
+
+  this.subscribe = function(...args) {
+    Logger.log('SubsCache - subscribe',args);
+    args.unshift(this.expireAfter);
+    return this.subscribeFor.apply(this, args);
+  }
+
+  this.subscribeFor = function(expireTime, ...args) {
+    var hash = EJSON.stringify(withoutCallbacks(args));
+
```

Comment:
Why would SSR compatibility be broken in these lines? Guidance appreciated, don't use SSR.